src/model.ts

```diff
@@ -0,0 +1,468 @@
+import { VDomModel } from '@jupyterlab/apputils';
+import {
+  ColorPalette,
+  ColorInterpolationSpace,
+  ColorRGBA64,
+  contrastRatio,
+  hslToRGB,
+  rgbToHSL,
+  ColorHSL
+} from '@microsoft/fast-colors';
+
+const black = new ColorRGBA64(0, 0, 0, 1);
+const white = new ColorRGBA64(1, 1, 1, 1);
+const threshold = 7;
+
+function isBlack(color: ColorRGBA64) {
+  let test = 'Undefined';
+  if (color.r === 0.0 && color.g === 0.0 && color.b === 0.0) {
+    test = 'True';
+  } else {
+    test = 'False';
+  }
+  return test;
+}
+
+function isWhite(color: ColorRGBA64) {
+  let test = 'Undefined';
+  if (color.r === 1.0 && color.g === 1.0 && color.b === 1.0) {
+    test = 'True';
+  } else {
+    test = 'False';
+  }
+  return test;
+}
+
+function hexToRGBA(h: string) {
+  let r = '0',
+    g = '0',
+    b = '0';
+
+  // 4 digits
+  if (h.length === 4) {
+    r = '0x' + h[1] + h[1];
+    g = '0x' + h[2] + h[2];
+    b = '0x' + h[3] + h[3];
+
+    // 6 digits
+  } else if (h.length === 7) {
+    r = '0x' + h[1] + h[2];
+    g = '0x' + h[3] + h[4];
+    b = '0x' + h[5] + h[6];
+  }
+  const a = '1';
+  const rgba = [Number(r) / 256, Number(g) / 256, Number(b) / 256, Number(a)];
+  return rgba;
+}
+
+function rgba_string_to_ColorRGBA64(colorValueRGBAstr: number[]) {
```

Comment:
In JavaScript the code convention is to use camel case

```suggestion
function rgbaStringToColorRGBA64(colorValueRGBAstr: number[]) {
```